lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -50,7 +50,7 @@
     <% mode_class = '' %>
   % endif
 <div class="course-container">
- % if course_program_info and course_program_info['category']=='xseries':
+ % if course_program_info and course_program_info.get('category')=='xseries':
```

Comment:
I was trying to fix unit tests but I can see how this change also prevents problems accessing dictionary values. I think this is a place for defensive coding like this. What do you think?